src/drivers/riscv_plic0.c

```diff
@@ -30,36 +30,45 @@
 extern metal_interrupt_handler_t __metal_global_interrupt_table[];
 
 static __inline__ unsigned int __metal_plic0_claim_interrupt(uint32_t hartid) {
-    return PLIC_REGW(PLIC_CONTEXT_BASE(hartid) +
-                     METAL_RISCV_PLIC0_CONTEXT_CLAIM);
+    return PLIC_REGW(
```

Comment:
Would be easier to read if there were a macro which took the hartid and returned the context address, or maybe a macro that took a hartid and returned the context claim register address? Something to avoid duplicating this computation several times.